contracts/SnappAuction.sol

```diff
@@ -68,13 +70,39 @@ contract SnappAuction is SnappBase {
         auctions[auctionIndex].size++;
     }
 
+    function applyAuction(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 _newStateRoot,
+        bytes32 _orderHash,
+        uint128[MAX_TOKENS] memory prices,
+        uint128[2*AUCTION_BATCH_SIZE] memory volumes
```

Comment:
Here I'm wondering if it would make sense to encode the volumes as bytes rather than as a vector like this. We are wasting 7 bytes per order (28bits * 2), which adds up to ~500k gas.

While we haven't done any gas optimisation anywhere, I'm wondering what the advantage of having the volumes and prices decompressed in a readable format like this is.
Otherwise, I'd argue we should just squash all of it into a single blob

Reply:
Thats cool with me. Shall we use bytes? This will require inclusion of a `BytesLibrary` (similar to that used in our Plasma contract). What better time than now?

Reply:
yeah, lets do it

Reply:
Great! I found that the BytesLib supplier, who wS only publishing to ethpm before, now has an npm version of `BytesLib.sol`. Even though I prefer ethpm, at least we can stay consistent now!

Reply:
[check it out!](https://github.com/GNSPS/solidity-bytes-utils)

Reply:
Why not use the solidity built in `bytes` type? https://solidity.readthedocs.io/en/v0.5.3/types.html#dynamically-sized-byte-array

Reply:
Well ya, this is the datatype we are going to use, but if there is any reason to do any byte-manipulation (such as concatenation or slicing), then I would be using the BytesLib to do so. However, I suppose your point is that we don't need to do any bit-manipulation in this context.

Reply:
We are going to have to assert some kind of length on these bytes (i.e. extra conditions for solution submissions). Is this all cool with everyone?

Reply:
Turns out, there is no need to validate price and volume sizes since any incorrect submissions can be challenged and fail.